### PR TITLE
Github Actions Create Release

### PR DESCRIPTION
**Pull request by mllemango**
### WHY are these changes introduced?

Current release process is cumbersome and can be automated by Github Actions!  

### WHAT is this pull request doing?

Automatically detects when a tag is created and creates a release with `.deb` and `.rpm` files attached.
Testing workflows is difficult, thus check https://github.com/mllemango/shopify-app-cli/releases/tag/v1.4.19 for result.

Note on local testing:
- use https://github.com/nektos/act
- had to use alternative runner image (`act -P ubuntu-latest=nektos/act-environments-ubuntu:18.04`)

### Update checklist

- [ ] I've added a CHANGELOG entry for this PR

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    - name: Create Release
+      id: create_release
+      uses: actions/create-release@v1
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      with:
+        tag_name: v${{ env.version }}
+        release_name: Release ${{ env.version }}
+        body: |
+          Changes in this Release
```
Comment (andyw8) [inline .github/workflows/release.yml:39]: I think we can use a generic message (see https://github.com/Shopify/shopify-app-cli/releases) so we won't need to worry about identifying the particular changes.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    steps:
+    - uses: actions/checkout@v2
+    
+    - name: Set up Ruby
+      uses: actions/setup-ruby@v1.1.2
+      with:
+        ruby-version: 2.6
+    
+    - name: Install dependencies
+      run: bundle install
```
Comment (paulomarg) [inline .github/workflows/release.yml:20]: I think we'll need one more step after this, to install `rpm` and (maybe?) `dpkg-deb` so we can actually build those packages. We have some docs on the packaging process:
- https://github.com/Shopify/shopify-app-cli/tree/master/packaging/debian
- https://github.com/Shopify/shopify-app-cli/tree/master/packaging/rpm

Comment (mllemango) [reply .github/workflows/release.yml:20]: I found from local testing that it wasn't necessary 🤨 but will keep an eye out for any issues surrounding this!

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        draft: true
+        prerelease: false
+    
+    - name: Upload Release Asset
+      id: upload-release-asset 
+      uses: actions/upload-release-asset@v1
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      with:
+        upload_url: ${{ steps.create_release.outputs.upload_url }} # This pulls from the CREATE RELEASE step above, referencing it's ID to get its outputs object, which include a `upload_url`. See this blog post for more info: https://jasonet.co/posts/new-features-of-github-actions/#passing-data-to-future-steps
```
Comment (andyw8) [inline .github/workflows/release.yml:51]: Let's format this line to avoid it being overly long?

File: .github/workflows/release.yml (hunk 2 above)
Comment (paulomarg) [reply .github/workflows/release.yml:20]: Cool, those packages were needed for mac (obviously) so I expected `dpkg` to be present for ubuntu, but good to know that `rpm` is there by default too!

**Comment by mllemango** [comment]
The release can also be created as a draft, let me know if that's more preferable than automatically publishing.

File: RELEASING.md [hunk 4]
```diff
@@ -49,7 +49,12 @@
     * commit the change and create a PR on the [Shopify Homebrew repository](https://github.com/Shopify/homebrew-shopify)
     * when PR is approved, merge into main branch
 
-1. Go to [releases](https://github.com/Shopify/shopify-app-cli/releases) page of `Shopify/shopify-app-cli` repo and create a new release:
+1. Check the "Actions" tab to see if the _Create Release_ workflow is successful. 
+   The workflow will automatically create a release with the latest tag and the `.deb`, `.rpm` assets attached.
+   
+   [Manual Instructions]
```
Comment (andyw8) [inline RELEASING.md:55]: We can remove the manual part. I'll still be in the git history should we ever need to refer back.

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+      run: |
+        cd packaging/builds
+        echo ::set-output name=name::$(ls)
+    
+    - name: Get Packages
+      id: binaries
+      run: |
+        cd packaging/builds/${{ steps.folder.outputs.name }}
+        echo $PWD
+        find . -type f -name "*.deb"
```
Comment (mkevinosullivan) [inline .github/workflows/release.yml:38]: This seems to be a duplicate line that's not really doing anything?

File: .github/workflows/release.yml [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+      id: folder
+      run: |
+        cd packaging/builds
+        echo ::set-output name=name::$(ls)
+    
+    - name: Get Packages
+      id: binaries
+      run: |
+        cd packaging/builds/${{ steps.folder.outputs.name }}
+        echo $PWD
```
Comment (andyw8) [inline .github/workflows/release.yml:38]: Was this just for debugging?

Comment (mllemango) [reply .github/workflows/release.yml:38]: Oh good catch 😅 it is, i'll remove

File: .github/workflows/release.yml [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+        deb=$(find . -type f -name '*.deb')
+        rpm=$(find . -type f -name "*.rpm")
+        echo ::set-output name=deb_file::"${deb:2}"
+        echo ::set-output name=rpm_file::"${rpm:2}"
+
+    - name: Create Release # https://hub.github.com/hub-release.1.html
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      run: |
+        hub release create ${{ steps.tag.outputs.value }} \
```
Comment (andyw8) [inline .github/workflows/release.yml:49]: Could we `cd` into `/home/runner/work/shopify-app-cli/shopify-app-cli/packaging/builds/${{ steps.folder.outputs.name }}` first?

Comment (mllemango) [reply .github/workflows/release.yml:48]: I don't see why not! will give that a try

File: .github/workflows/release.yml (hunk 5 above)
Comment (mllemango) [reply .github/workflows/release.yml:38]: 🤦‍♀️ sorry, this is also a debugging line

File: RELEASING.md [hunk 8]
```diff
@@ -49,15 +49,5 @@
...
-      ```
-      Release of version X.Y.Z of Shopify App CLI
-    
-      Please refer to [CHANGELOG](https://github.com/Shopify/shopify-app-cli/blob/master/CHANGELOG.md) for details.
-      ```
-    * upload the `.deb` and `.rpm` files from `packaging/builds/X.Y.Z/` (generated in step 9)
-    * if it's a pre-release version, select the "This is a pre-release" checkbox
-    * and click "Publish release".
+1. Check the "Actions" tab to see if the _Create Release_ workflow is successful. 
+   The workflow will automatically create a release with the latest tag and the `.deb`, `.rpm` assets attached.
```
Comment (paulomarg) [inline RELEASING.md:53]: We should mention the homebrew update here as well.

Comment (mllemango) [reply RELEASING.md:53]: Currently on it 😄

**Review by andyw8 — APPROVED** [review]

**Pull request merged**